archive-links.py

```diff
@@ -0,0 +1,67 @@
+"""
+Script to ensure that all bookmarks from raindrop have been archived in the Way Back Machine (https://archive.org/web/)
+"""
+import argparse
+from typing import List, Callable, Tuple
+import json
+import sys
+from raindrop_api import extract_bookmarks
+from wayback_api import check_and_archive_url
+import logging
+
+DEFAUlT_PAGE_SIZE = 40
+PRESTERITY_COLLECTION_ID = 2021037
+
+def build_parser() -> argparse.ArgumentParser:
+	"""Construct argument parser for script.
+
+	:return: ArgumentParser
+	"""
+	parser = argparse.ArgumentParser(description=__doc__)
+	help_msg = 'id of Raindrop collection of interest; default is {0} (Presterity)'.format(
+	    PRESTERITY_COLLECTION_ID)
+	parser.add_argument('--collection', '-c', type=int, default=PRESTERITY_COLLECTION_ID, 
+	                    help=help_msg)
+	help_msg = 'number of requested bookmarks per query; default is {0} (max allowed)'.format(
+	    DEFAUlT_PAGE_SIZE)
+	parser.add_argument('--page-size', '-s', type=int, default=DEFAUlT_PAGE_SIZE, help=help_msg)
+	parser.add_argument('--outfile', '-o', help='file for results; default is stdout')
+	parser.add_argument('--verbose', '-v', action='store_true', help='log level DEBUG')
+	return parser
+
+
+def archive_urls(urls: List[str], log, cb: Callable[[dict], None]) -> None:
+	"""
+	Batch archive a list of URLs
+
+	:param urls: list of URLs to archive
+	:param cb: a callback that is executed each time a URL is successfully archived. 
+	The callback is given the argument: {original: <url>, archive: <url>}
+	"""
+	for url in urls:
+		archived = check_and_archive_url(url)
+		if archived is None:
+			log.warn("unable to archive url %s", url)
+			data = {'original': url}
+		else:
+			data = {'original': url, 'archive': archived}
+		cb(data)
+
+if __name__ == '__main__':
+	args = build_parser().parse_args()
+	logging.basicConfig(level=logging.WARN)
+	log = logging.getLogger(__name__)
+	if args.verbose:
+		log.setLevel(logging.DEBUG)
+	else:
+		log.setLevel(logging.INFO)
+
```

Comment:
Having log at the module level is actually a Python logging best practice. In non-script modules, you'll often see it declared right after import logging, and set to the logger for `__name__`. That works nicely given the scoped names of Python modules; setting the log level for `foo` also sets it for `foo.*`, `foo.*.*`, etc. I know it seems messy to have it at the module level; it takes some getting used to.